src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java

```diff
@@ -46,18 +36,20 @@ public Query buildConceptChildrenQuery(
       SnapshotBuilderDomainOption domainOption, int parentConceptId) {
 
     // concept table and its fields concept_name and concept_id
-    TableVariable concept = TableVariable.forPrimary(TablePointer.fromTableName(CONCEPT));
-    FieldVariable conceptName = concept.makeFieldVariable(CONCEPT_NAME);
-    FieldVariable conceptId = concept.makeFieldVariable(CONCEPT_ID);
-    FieldVariable conceptCode = concept.makeFieldVariable(CONCEPT_CODE);
+    Concept concept = new Concept();
+    FieldVariable conceptName = concept.name();
+    FieldVariable conceptId = concept.concept_id();
+    FieldVariable conceptCode = concept.code();
 
     // concept_ancestor joined on concept.concept_id = ancestor_concept_id.
     // We use concept_ancestor for the rollup count because we want to include counts
     // from all descendants, not just direct descendants.
-    TableVariable conceptAncestor =
-        TableVariable.forJoined(
-            TablePointer.fromTableName(CONCEPT_ANCESTOR), ANCESTOR_CONCEPT_ID, conceptId);
-    FieldVariable descendantConceptId = conceptAncestor.makeFieldVariable(DESCENDANT_CONCEPT_ID);
+
+    ConceptAncestor conceptAncestor =
+        new ConceptAncestor(
+            new TableVariableBuilder().join(ConceptAncestor.ANCESTOR_CONCEPT_ID).on(conceptId));
```

Comment:
Can this form can be pushed down to the table class?
```
    ConceptAncestor conceptAncestor = new ConceptAncestor(conceptId);
```
Assuming that we always join through `ConceptAncestor.ANCESTOR_CONCEPT_ID`. If we also join through descendent then maybe we'd need both forms

Reply:
It's joined on both `ANCESTOR_CONCEPT_ID` and `DESCENDENT_CONCEPT_ID`. I think i'd rather prefer having it in the builder class to remove from creating more constructors? What are your thoughts on this? 

A possible solution could be making a constructor:

```
  public ConceptAncestor(String joinField, FieldVariable joinOnParent) {
    super(
        tablePointer,
        joinField,
        joinOnParent
        false
        )
  }
  
 // calls would be 
 ConceptAncestor conceptAncestor = new ConceptAncestor(ConceptAncestor.ANCESTOR_CONCEPT_ID, conceptId)
// or 
 ConceptAncestor conceptAncestor = new ConceptAncestor(ConceptAncestor.DESCENDENT_CONCEPT_ID, conceptId)
```

But, this adds more constructors and removes the abstract nature of the builder pattern.